### PR TITLE
Add libsndfile

### PR DESCRIPTION
**Pull request by SirLynix**
This is going to be fine...
From #203

### CODE REVIEW

**Comment by waruqi** [comment]
maybe you need pass -DFLAC__NO_DLL to cmake in libsndfile

**Comment by SirLynix** [comment]
It's already there: <https://github.com/xmake-io/xmake-repo/pull/213/checks?check_run_id=1557192128#step:4:163>

File: packages/l/libsndfile/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+
+        -- It seems libsndfile expects CMAKE_MSVC_RUNTIME_LIBRARY
+        if package:is_plat("windows") then
+            if not package:config("shared") then
+                -- libsndfile doesn't build well with a static libFLAC, this fixes it
+                local cmake = assert(io.open("CMakeLists.txt", "a"))
+                cmake:write([[
+if (WIN32 AND NOT BUILD_SHARED_LIBS)
```
Comment (waruqi) [inline packages/l/libsndfile/xmake.lua:28]: we need not add it, we can use lua to handle it

File: packages/l/libsndfile/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        table.insert(configs, "-DBUILD_EXAMPLES=OFF")
+        table.insert(configs, "-DBUILD_TESTING=OFF")
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+
+        -- It seems libsndfile expects CMAKE_MSVC_RUNTIME_LIBRARY
+        if package:is_plat("windows") then
+            if not package:dep("libflac"):config("shared") then
+                -- libsndfile doesn't build well with a static libFLAC, this fixes it
+                local cmake = assert(io.open("CMakeLists.txt", "a"))
```
Comment (waruqi) [inline packages/l/libsndfile/xmake.lua:26]: we need not add assert, xmake will check the return value of `io.open` automatically.

File: packages/l/libsndfile/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                cmake:write("add_definitions(-DFLAC__NO_DLL)\n")
+                cmake:close() -- Flush the file
```
Comment (waruqi) [inline packages/l/libsndfile/xmake.lua:28]: remove this comment

**Pull request merged**